lib/chef/resource/helpers/path_helpers.rb

```diff
@@ -0,0 +1,39 @@
+require "chef-utils/dist" unless defined?(ChefUtils::Dist)
+class Chef
+  module ResourceHelpers
+    # Helpers for path manipulation
+    module PathHelpers
+      extend self
+      # The habitat binary path for Infra Client
+      # @return [String]
+      def hab_binary_path
```

Comment:
This name is misleading, reading it I'd expect it to be looking for the habitat cli binary path and not chef infra binary path. recommend something like chef-client_hab_bin_path.

We also should not assume latest.  We should either enforce bin-linking or require bin linking for this to work and simplify to using /usr/bin/chef-client. This would break the advantage of having multiple versions installed and being able to test latest version along side previous version but still default to previous version till ready to migrate to latest version as default.

Reply:
Yeah I have also been having same thoughts but finally resorted to traversal of version for reliability, because my concern is the binlink is not created for a habitat package by default unless the user specifies `--binlink` option when installing it via `hab pkg install` so we cannot assume binlink being present that will break the resources as there are multiple resources which have property whose default value should point to chef-client's binary executable.
About the naming, makes sense. I have updated it. Thanks.

Reply:
would `hab pkg path chef/chef-infra-client` give the latest version or the active version of a package. If it gives the active version then that should work.

Biggest problem we have to figure out how they would test a new version release along side desired production release version or make note in the resources that it'll always default to latest and stop promoting testing new versions of chef-client along side current active version as a feature.

Reply:
@Stromweld  `hab pkg path chef/chef-infra-client` returns the latest version based on semantic version. So in case multiple version installation it picks up latest version and within same version, the version with latest timestamp, which is what I'm emulating when deciding the binary path.

Regarding a new property for version on the resources, the existing `chef_binary_path` property itself can be used. One can always specify the entire binary path using the {version}/{timestamp} identifier e.g
```
chef_client_cron do
  chef_binary_path "/hab/pkgs/chef/chef-infra-client/19.1.0/20230822151044/bin/chef-client"
end
```
Hope that answers your queries?

Reply:
That would work as a work around, but for user experience we should make it easier. It makes sense to default to latest installed if not specified but if we add an option for version and can create that same path with simple string interpolation. something like "/hab/pkgs/chef/chef-infra-client/#{client-builder-version}/bin/chef-client"

I'd also be happy to work on that as a separate PR.

Reply:
hmm, sure.